optimize_build.sh

```diff
@@ -1,18 +1,44 @@
+#!/bin/bash
+set -e
+
+BINARYEN_VERS=110
+BINARYEN_DWN="https://github.com/WebAssembly/binaryen/releases/download/version_${BINARYEN_VERS}/binaryen-version_${BINARYEN_VERS}-x86_64-linux.tar.gz"
+
 mkdir -p artifacts
 cargo fmt --all
 cargo clippy --fix
 cargo clean
-RUSTFLAGS='-C link-arg=-s' cargo build  --workspace --exclude test-utils --release --lib --target wasm32-unknown-unknown
+
+# Install toolchains
+# cargo add target wasm32-unknown-unknown
+rustup target add wasm32-unknown-unknown
+cargo install cosmwasm-check
+if ! which wasm-opt; then
+  curl -OL $BINARYEN_DWN
+  tar xf binaryen-version_${BINARYEN_VERS}-x86_64-linux.tar.gz
+  export PATH=$PATH:$PWD/binaryen-version_${BINARYEN_VERS}/bin
+fi
+
+# Check toolchain version to be: rust 1.69.0 and wasm-opt 110
+WASMOPT_VERSION=$(wasm-opt --version | awk '{print $3}')
+RUSTC_VERSION=$(rustc -V | awk '{print $2}')
+if [ "$RUSTC_VERSION" != "1.69.0" ] || [ "$WASMOPT_VERSION" != "110" ]; then   
+  echo -e "\n ** Warning: The required versions for Rust and wasm-opt are 1.69.0 and 110, respectively. Building with different versions may result in failure.\n"
+fi
+
```

Comment:
we can declare required rustc version and wasm-opt version as well at top and compare with it . We might need to update these in future so it will be easy.

Reply:
The required versions have been declared at the top of the script now.